samples/client/petstore/go/go-petstore/enum_class.go

```diff
@@ -10,5 +10,12 @@
 
 package petstore
 
-type EnumClass struct {
-}
+type EnumClass string
+
+// List of EnumClasss
+const (
+	_abc EnumClass = "_abc"
```

Comment:
The generator would use them. I think what is being suggested is to include a similar override in the GoClientCodegen.java and GoServerCodegen.java

Just briefly reading, something like this may help.
```
    @Override
    public String toEnumVarName(String name, String datatype) {
...
        String enumName = sanitizeName(underscore(name).toUpperCase());
        enumName = enumName.replaceFirst("^_", "");
        enumName = enumName.replaceFirst("_$", "");
...
    }
```